### ISSUE TITLE
Speed Up our Docker Travis CI builds

### ISSUE THREAD

**Issue by eldadfux** [appwrite/appwrite#714]
We want to use caching to speed up Appwrite's Travis CI build process, and we can use the community help here.

Currently our build process time is around ~10 minutes which is OK, but as faster it can be less time maintainers needs to wait for confirmation that there changes are running as expected.

This change should be focused on our Travis CI YAML file. Any suggestions for improving the code base should be handled on an independent issue.

**Comment by SoftCreatR** [comment]
1. Step: Get rid of Travis and use GitHub Actions ;)

**Comment by enggal64** [comment]
I

**Comment by Sj1218305** [comment]
I want to claim this issue.
@eldadfux , is it still open?

**Comment by eldadfux** [comment]
@Sj1218305 Yes it is, please reach out on Discord to share your plan of action on this issue.

**Comment by eldadfux** [comment]
@Sj1218305 please work against branch `0.7.x` which is where the upcoming Appwrite version located

**Comment by Glide7** [comment]
Hey @Sj1218305 :)
Are you still working on this issue? Do you perhaps need help from the team?

**Comment by Sj1218305** [comment]
@Glide7 , I am working on this. Actually , I am new to Docker so I need some more time. I will surely reach out if any kind of help is required.

**Comment by Glide7** [comment]
Awesome, thanks for the update!

**Comment by Glide7** [comment]
@Sj1218305 also, feel free to join the #mainteners channel on our discord! https://discord.gg/yVQ7RBv4

**Comment by Glide7** [comment]
Hey @Sj1218305 - how are things going? Do you need assistance from us?

**Comment by hack3r-0m** [comment]
hi, I would like to take it.
@SoftCreatR suggested I will use this approach https://github.com/marketplace/actions/build-docker-images-using-cache#from-a-compose-file. I am thinking of creating a new branch from 0.7.x and make PR to that so it does not disturb the current development flow. I have a few doubts:
- if I am not wrong, there is no image of 0.7.x on the docker hub
-  in the `compose_file: docker-compose.yml` line, is the path to docker-compose.yml relative considering .github/workflows working dir?
- that action has some info regarding [testing](https://github.com/marketplace/actions/build-docker-images-using-cache#tests), is it for that particular repo, or is it relevant in this case?

**Comment by eldadfux** [comment]
@hack3r-0m moving to Github actions is not part of this issue scope. The team is currently using Travis CI for ~60 repos and we need to have a wider plan for this kind of migration, which unfortunately is not a priority right now.

**Comment by hack3r-0m** [comment]
@eldadfux once 0.7.x is ready and pushed to docker, let say new PRs get merged into 0.7.x branch then how is old image updated on docker hub, there must be something step in .travis.yml file to merge new code into existing Docker image? As far as caching Travis CI, [docs](https://docs.travis-ci.com/user/caching/#things-not-to-cachel) recommend not to cache images, so the only thing to cache is `docker-compose up -d`(containers)? btw, is there a need to install docker explicitly in travis build as `language: minimal` comes with the latest docker version([source](https://docs.travis-ci.com/user/languages/minimal-and-generic/#minimal))?

**Comment by eldadfux** [comment]
@hack3r-0m sorry for the late reply. Not sure about the Docker part, can you verify its already installed in `language: minimal`?

**Comment by hack3r-0m** [comment]
@eldadfux language has been changed to `language: shell` (as per docs, it is same as minimal and has docker. [source](https://docs.travis-ci.com/user/languages/minimal-and-generic/#minimal)) recently and also deploy.sh has been added in 0.7.x, is this issue still relevant?

**Comment by eldadfux** [comment]
@hack3r-0m yes, mind submitting a PR removing Docker so we can make sure tests are still green?

**Comment by hack3r-0m** [comment]
sure, i will try to do it ASAP

**Comment by hack3r-0m** [comment]
@eldadfux `buildx` (19.03.8)not supported with default docker version which comes pre-installed, while their docs say it is available on 19.03+

**Comment by slcnx** [comment]
Why don't you try to put construction phase to Dockerfile , so that we can use the docker cache

**Comment by eldadfux** [comment]
> Why don't you try to put construction phase to Dockerfile , so that we can use the docker cache

We actually do use a multi-stage build, but we build the image from source on the CI to make sure it picks up all the changes made by the team and community.

**Comment by StarTrooper08** [comment]
@eldadfux I would like to work on this issue
